modules/renter/downloadcache.go

```diff
@@ -0,0 +1,53 @@
+package renter
+
+import "time"
+
+// addChunkToCache adds the chunk to the cache if the download is a streaming
+// endpoint download.
+func (udc *unfinishedDownloadChunk) addChunkToCache(data []byte) {
+	if udc.download.staticDestinationType == destinationTypeSeekStream {
```

Comment:
anti-pattern - instead of having an `if` block that wraps the whole function, have an inverse that returns early.

```
if udc.download.staticDestinationType != destinationTypeSeekStream {
    // some comment explaining why we only cache destinationTypeSeekStream
    return
}
```